optuna/samplers/_tpe/sampler.py

```diff
@@ -155,13 +182,62 @@ def infer_relative_search_space(
         self, study: Study, trial: FrozenTrial
     ) -> Dict[str, BaseDistribution]:
 
-        return {}
+        if not self._multivariate:
+            return {}
+
+        search_space = {}  # type: Dict[str, BaseDistribution]
+        for name, distribution in self._search_space.calculate(study).items():
+            if not isinstance(
+                distribution,
+                (
+                    distributions.UniformDistribution,
+                    distributions.LogUniformDistribution,
+                    distributions.DiscreteUniformDistribution,
+                    distributions.IntUniformDistribution,
+                    distributions.IntLogUniformDistribution,
+                    distributions.CategoricalDistribution,
+                ),
+            ):
+                continue
```

Comment:
Sounds reasonable to use `distributions.DISTRIBUTION_CLASSES`. I will fix this as you suggested.

Reply:
Just for clarification, can't we just skip unsupported distributions and pass them to the `sample_independent`?

Reply:
That's also fine. In that case, it might be a good idea to let users/us know when we're falling back to independent similar to [how it's done in the `CmaEsSampler`](https://github.com/optuna/optuna/blob/master/optuna/samplers/_cmaes.py#L374-L377) (note that there might be user-defined distributions).

Reply:
Thank you, raising a warning like `CmaEsSampler` seems more reasonable to me.